test/e2e/common/util_cleanup.go

```diff
@@ -106,34 +148,68 @@ func DeleteAllVolumeResources() (bool, bool) {
 	msvs, err := gTestEnv.DynamicClient.Resource(msvGVR).Namespace("mayastor").List(context.TODO(), metav1.ListOptions{})
 	if err != nil {
 		// This function may be called by AfterSuite by uninstall test so listing MSVs may fail correctly
-		logf.Log.Info("DeleteAllVolumeResources: list MSVs failed.", "Error", err)
+		logf.Log.Info("DeleteAllMsvs: list MSVs failed.", "Error", err)
 	}
 	if err == nil && msvs != nil && len(msvs.Items) != 0 {
-		logf.Log.Info("DeleteAllVolumeResources: deleting MayastorVolumes")
 		for _, msv := range msvs.Items {
-			if err := DeleteMSV(msv.GetName()); err != nil {
-				success = false
+			logf.Log.Info("DeleteAllMsvs: deleting MayastorVolume", "MayastorVolume", msv.GetName())
+			if delErr := DeleteMSV(msv.GetName()); delErr != nil {
+				logf.Log.Info("DeleteAllMsvs: failed deleting MayastorVolume", "MayastorVolume", msv.GetName(), "error", delErr)
 			}
 		}
 	}
 
 	// Wait 2 minutes for resources to be deleted
+	numMsvs := 0
 	for attempts := 0; attempts < 120; attempts++ {
-		numMsvs := 0
 		msvs, err := gTestEnv.DynamicClient.Resource(msvGVR).Namespace("mayastor").List(context.TODO(), metav1.ListOptions{})
 		if err == nil && msvs != nil {
 			numMsvs = len(msvs.Items)
-		}
-		if numMsvs == 0 {
-			break
+			if numMsvs == 0 {
+				break
+			}
 		}
 		time.Sleep(1 * time.Second)
 	}
+	logf.Log.Info("DeleteAllMsvs:", "number of MayastorVolumes", numMsvs)
+
+	return numMsvs, err
+}
+
+func DeleteAllPoolFinalizers() (bool, error) {
+	deletedFinalizer := false
 
-	return success, foundResources
+	poolGVR := schema.GroupVersionResource{
+		Group:    "openebs.io",
+		Version:  "v1alpha1",
+		Resource: "mayastorpools",
+	}
+
+	pools, err := gTestEnv.DynamicClient.Resource(poolGVR).Namespace("mayastor").List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		logf.Log.Info("DeleteAllPoolFinalisers: list MSPs failed.", "Error", err)
+		return false, err
+	} else if len(pools.Items) != 0 {
+		for _, pool := range pools.Items {
+			empty := make([]string, 0)
+			logf.Log.Info("DeleteAllPoolFinalizers", "pool", pool.GetName())
+			finalizers := pool.GetFinalizers()
+			if finalizers != nil {
+				logf.Log.Info("Removing all finalizers", "pool", pool.GetName(), "finalizer", finalizers)
+				pool.SetFinalizers(empty)
+				_, err = gTestEnv.DynamicClient.Resource(poolGVR).Namespace("mayastor").Update(context.TODO(), &pool, metav1.UpdateOptions{})
+				if err != nil {
```

Comment:
Looks like err gets passed back to the caller only if it is the last pool that errors when updated?  Maybe you need a separate variable as in the other functions like this.

Reply:
Well spotted!
I don't think accumulating the errors is useful but recording that an error occurred is.
I will set it to the last error encountered